test/kernel_acl.js

```diff
@@ -88,28 +91,62 @@ contract('Kernel ACL', accounts => {
 
         it('root cannot grant permission', async () => {
             return assertRevert(async () => {
-                await kernel.grantPermission(granted, app, role, granted, { from: permissionsRoot })
+                await kernel.grantPermission(granted, app, role, { from: permissionsRoot })
+            })
+        })
+
+        context('transferring ownership', () => {
+            const newOwner = accounts[8]
+
+            beforeEach(async () => {
+                const receipt = await kernel.setPermissionOwner(newOwner, app, role, { from: granted })
+                assertEvent(receipt, 'ChangePermissionOwner')
+            })
+
+            it('changes owner', async () => {
+                const owner = await kernel.getPermissionOwner(app, role)
+                assert.equal(owner, newOwner, 'owner should have changed')
+            })
+
+            it('can grant permission', async () => {
+                const receipt = await kernel.grantPermission(newOwner, app, role, { from: newOwner })
+                assertEvent(receipt, 'SetPermission')
+            })
+
+            it('fails when setting owner to the zero address', async () => {
+                return assertRevert(async () => {
+                    await kernel.setPermissionOwner('0x00', app, role, { from: newOwner })
+                })
+            })
+
+            it('old owner lost power', async () => {
+                return assertRevert(async () => {
+                    await kernel.grantPermission(newOwner, app, role, { from: granted })
+                })
             })
         })
 
         context('self-revokes permission', () => {
             beforeEach(async () => {
-                await kernel.revokePermission(granted, app, role, { from: granted })
+                const receipt = await kernel.revokePermission(granted, app, role, { from: granted })
+                assertEvent(receipt, 'SetPermission')
             })
 
             it('can no longer perform action', async () => {
                 assert.isFalse(await kernel.canPerform(granted, app, role))
             })
 
-            it('permissions root can re-create', async () => {
-                await kernel.createPermission(granted, app, role, granted, { from: permissionsRoot })
-                assert.isTrue(await kernel.canPerform(granted, app, role))
+            it('permissions root cannot re-create', async () => {
+                return assertRevert(async () => {
+                    await kernel.createPermission(granted, app, role, granted, { from: permissionsRoot })
+                })
```

Comment:
I'd add a test here to make sure that `granted` could still re-grant itself the permission it owns.

Reply:
Added https://github.com/aragon/aragon-core/pull/156/commits/8865a1a16776bed9090a2c2e1224a742cd52a005